custom/enikshay/reports/historical_adherence.py

```diff
@@ -240,7 +243,7 @@ def _source_is_enikshay(case):
             return case.dynamic_case_properties().get('adherence_source') in ('enikshay', '')
 
         return sorted(
-            adherence_cases, key=lambda c: (_source_is_enikshay(c), c.modified_on)
+            adherence_cases, key=lambda c: (_source_is_enikshay(c), not c.closed, c.modified_on, c.opened_on)
```

Comment:
If a second adherence for a given day is created through the app, the previously open case will be closed. Therefore both cases will have the same `modified_on` value, so we need to defer to `opened_on` to determine which case is older. We can't simply rely on open/close status because the cases could later both be closed by a periodic task that closes adherence cases older than some threshold.